linkerd/app/core/src/control.rs

```diff
@@ -19,8 +33,48 @@ impl fmt::Display for ControlAddr {
     }
 }
 
+type BalanceBody =
+    http::balance::PendingUntilFirstDataBody<tower::load::peak_ewma::Handle, http::glue::Body>;
+
+type RspBody = linkerd2_http_metrics::requests::ResponseBody<BalanceBody, classify::Eos>;
+
+pub type Client<B> = linkerd2_buffer::Buffer<http::Request<B>, http::Response<RspBody>>;
+
+impl Config {
+    pub fn build<B, I>(
+        self,
+        dns: dns::Resolver,
+        metrics: ControlHttpMetrics,
+        identity: tls::Conditional<I>,
+    ) -> Client<B>
+    where
+        B: http::HttpBody + Send + 'static,
+        B::Data: Send,
+        B::Error: Into<Error> + Send + Sync,
+        I: Clone + tls::client::HasConfig + Send + 'static,
+    {
+        svc::connect(self.connect.keepalive)
+            .push(tls::ConnectLayer::new(identity))
+            .push_timeout(self.connect.timeout)
+            .push(self::client::layer())
+            .push(reconnect::layer({
+                let backoff = self.connect.backoff;
+                move |_| Ok(backoff.stream())
+            }))
+            .push_spawn_ready()
+            .push(self::resolve::layer(dns))
+            .push_on_response(self::control::balance::layer())
+            .push(metrics.into_layer::<classify::Response>())
+            .push(self::add_origin::Layer::new())
+            .into_new_service()
+            .check_new_service()
+            .push_on_response(svc::layers().push_spawn_buffer(self.buffer_capacity))
+            .new_service(self.addr)
```

Comment:
should we maybe stick an `instrument` in this stack somewhere, or does that happen elsewhere?